src/ray/gcs/gcs_client/service_based_accessor.cc

```diff
@@ -391,33 +400,50 @@ Status ServiceBasedNodeInfoAccessor::AsyncSubscribeToNodeChange(
     const StatusCallback &done) {
   RAY_LOG(DEBUG) << "Subscribing node change.";
   RAY_CHECK(subscribe != nullptr);
-  ClientTable &client_table = client_impl_->GetRedisGcsClient().client_table();
-  auto status = client_table.SubscribeToNodeChange(subscribe, done);
+  RAY_CHECK(node_change_callback_ == nullptr);
+  node_change_callback_ = subscribe;
+
+  auto on_subscribe = [this, subscribe](const std::string &id, const std::string &data) {
```

Comment:
There is no need to capture the `subscribe` as it is  not used.

Reply:
fixed